### DISCUSSION TITLE
How to implement {Seachbox} in header component, just like any other E-commerce.

### DISCUSSION THREAD

**Discussion by N-pharmarack** [algolia/instantsearch#5305] [Q&A] [answered]
I'm loving react-instantsearch , and I want to go one step ahead with it.

I'm making an e-commerce website, and just like any site, I want searchbox to be in header component, but I'm stuck on this part.

So if I submit query from my seachBar in header then only it should show me hits.
For ex. If I type 'apple', then onSubmit, it should redirect me to www.flipkart.com/search/?query="apple"

I'm using react-instantsearch-dom with TypesenseInstantSearchAdapter

It would be great if you guys direct me on how to achieve this use case of mine.

Thanks in advance.
![image](https://user-images.githubusercontent.com/95023092/162166782-c912c0d1-fbea-4a45-a81d-7f318c7c1025.png)

**Comment by Haroenv** [comment, marked as answer]
If you want InstantSearch in the header you have some choices:

1. a portal
2. make InstantSearch the more global wrapper of the page
3. use a regular input and communicate with InstantSearch using `searchState` (and having a virtual searchbox on the page)

**Reply by N-pharmarack** [reply]
Hello @Haroenv,
Thank you so much for replying.
Can you provide me some sandbox links, or some guides to follow??
And what do you mean by "A Portal" exactly?

**Reply by N-pharmarack** [reply]
Also I can't use custom input as I will be requiring Autocomplete in future useCases

**Reply by Haroenv** [reply]
What you actually want it seems is for the input not to be a searchbox, but a form which on submit goes to the /search page (which has InstantSearch) and routing set up so the query is read

**Reply by N-pharmarack** [reply]
As you see in the video I have done what you suggesting by writing this code 

```
const Header = () => {
  const [query,setQuery] = useState('');
  const handleChange = (event) => {
    setQuery(event.target.value);

  }
  const handleSumbit = (e) => {
    e.preventDefault();
    window.location.href = `http://${hostname}:${port}/list/search/?query=${query}`;
  }
  return (
    <>
      <h4>Search</h4>
      <form onSubmit={handleSumbit}>
        <input type="text" onChange={handleChange}/>
        <input type="submit" />
      </form>
      <h5>{query}</h5>
    </>
  );
}

export default Header;
```

What I want is to move the {SearchBox} component to Header so that in future I can implement Autocomplete.

Also Correct me if i'm wrong To hide the SeachBox  component in my current code, I need to use VirtualSearchBox, right??

https://user-images.githubusercontent.com/95023092/162176433-aacbe311-3ed8-4f6f-8c00-2014328914c9.mp4
